misk-redis/src/main/kotlin/misk/redis/Redis.kt

```diff
@@ -0,0 +1,55 @@
+package misk.redis
+
+import java.time.Duration
+
+/** A Redis client. */
+interface Redis {
+  /**
+   * Deletes a single key.
+   *
+   * @param key the key to delete
+   * @return 0 if the key was not deleted, 1 if the key was deleted
+   */
+  fun del(key: String): Long
+
+  /**
+   * Deletes multiple keys.
+   *
+   * @param keys the keys to delete
+   * @return 0 if none of the keys were deleted, otherwise a positive integer
+   *         representing the number of keys that were deleted
+   */
+  fun del(vararg keys: String): Long
+
+  /**
+   * Retrieves a key.
+   *
+   * @param key the key to retrieve
+   * @return a string value if the key was found, null if the key was not found
+   */
+  fun get(key: String): String?
+
+  /**
+   * Sets a value for a key.
+   *
+   * @param key the key to set
+   * @param value the value to set
+   * @return the value that was set
+   */
+  fun set(key: String, value: String): String
+
+  /**
+   * Sets a key with an expiration date.
+   *
+   * @param key the key to set
+   * @param expiryDuration the amount of time before the key expires
+   * @param value the value to set
+   * @return the value that was set
+   */
+  fun setex(key: String, expiryDuration: Duration, value: String): String
```

Comment:
make the expiration an optional parameter on `set`?

Reply:
I opted for having `set` be an operator, which means we can't have default variables in the function signature. What do you think about keeping `setex` a separate function?

Reply:
I would still just call it `set` - you can have overloaded methods in Kotlin.

Reply:
Changed `setex` to `set`